tests/core/CoreTest.py

```diff
@@ -514,7 +517,7 @@ def test_run_result_handler_exception(self):
         on_result = unittest.mock.Mock(side_effect=ValueError)
 
         with self.assertLogs(logging.getLogger()) as cm:
-            run({bear}, on_result)
+            self.execute_run({bear}, on_result)
```

Comment:
but if you want to supply a custom callback, really use `run` directly. `execute_run` was written to be able to retrieve the results for assertion. Putting a custom callback into `execute_run` turns this into a simple `run(bear, custom_callback)` call (which is done here)